packages/babel-plugin-proposal-object-rest-spread/src/index.js

```diff
@@ -240,6 +240,31 @@ export default declare((api, opts) => {
     }
   }
 
+  function hasMoreThanOneBinding(node) {
+    // ArrayPattern
+    if (node?.elements) {
+      if (node.elements.length > 1) return true;
+      else return hasMoreThanOneBinding(node.elements[0]);
+      // ObjectPattern
+    } else if (node?.properties) {
+      if (node.properties.length > 1) return true;
+      else return hasMoreThanOneBinding(node.properties[0]);
```

Comment:
we can early return `false` when `node.properties.length` is `0`, e.g.

```js
const {} = {}
```

Reply:
I'm stepping through this and am unable to trigger this case. I believe it is because `hasMoreThanOneBinding` is called within `visitRestElements` which means we must always have at least one property somewhere in the tree if there is a rest element.

I've added the guard for safety in case the calling pattern changes outside this function but I wasn't able to trigger this case after trialing many inputs with a debugger. Similarly I wasn't able to emulate the issue in https://astexplorer.net/ if I had at least one restElement somewhere on the LHS